browser/ui/toolbar/brave_app_menu_model.cc

```diff
@@ -284,18 +250,175 @@ void BraveAppMenuModel::InsertBraveMenuItems() {
                             IDS_APP_MENU_IPFS_PUBLISH_LOCAL_FOLDER,
                             keys_command_index);
     }
-    int index = IsCommandIdEnabled(IDC_SHOW_BRAVE_SYNC)
-                    ? GetIndexOfBraveSyncItem() + 1
-                    : GetLastIndexOfSecondSection();
+    const int index = GetNextIndexOfSecondSection();
     InsertSubMenuWithStringIdAt(index, IDC_APP_MENU_IPFS, IDS_APP_MENU_IPFS,
                                 &ipfs_submenu_model_);
 
     auto& bundle = ui::ResourceBundle::GetSharedInstance();
     const auto& ipfs_logo = *bundle.GetImageSkiaNamed(IDR_BRAVE_IPFS_LOGO);
     ui::ImageModel image_model = ui::ImageModel::FromImageSkia(ipfs_logo);
     SetIcon(index, image_model);
+    need_separator = true;
+  }
+#endif
+
+  if (need_separator) {
+    InsertSeparatorAt(GetNextIndexOfSecondSection(), ui::NORMAL_SEPARATOR);
+  }
+}
+
+void BraveAppMenuModel::BuildThirdSection() {
+  // First, reorder original menus We want to move them in order of bookmark,
+  // download and extensions.
+  absl::optional<size_t> bookmark_item_index =
+      GetIndexOfCommandId(IDC_BOOKMARKS_MENU);
+
+  // If bookmark is not used, we don't need to adjust download item.
+  if (bookmark_item_index.has_value()) {
+    // Place download menu under bookmark.
+    DCHECK(IsCommandIdEnabled(IDC_SHOW_DOWNLOADS));
+    RemoveItemAt(GetIndexOfCommandId(IDC_SHOW_DOWNLOADS).value());
+    InsertItemWithStringIdAt(bookmark_item_index.value(), IDC_SHOW_DOWNLOADS,
+                             IDS_SHOW_DOWNLOADS);
+  }
+
+  if (IsCommandIdEnabled(IDC_MANAGE_EXTENSIONS)) {
```

Comment:
**Note:** There is a Chromium feature for this, and it also adds it in the 2023 refresh - I wonder if we should add a `DCHECK` that it doesn't already exist.